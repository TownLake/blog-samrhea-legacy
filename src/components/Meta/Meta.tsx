import React from "react";

interface Props {
  description: string;
  image?: string;
  title: string;
}

const Meta: React.FC<Props> = ({ description, title, image }: Props) => (
  <>
    <title>{title}</title>
    <meta name="description" content={description} />
    <meta name="twitter:card" content="summary_large_image" />
    <meta name="twitter:title" content={title} />
    <meta name="twitter:description" content={description} />
    <meta name="og:title" content={title} />
    <meta name="og:type" content="website" />
    <meta name="og:description" content={description} />

    {image ? (
      <>
        <meta name="image" content={image} />
        <meta name="og:image" content={image} />
        <meta name="twitter:image" content={image} />
      </>
    ) : null}
  </>
);

export default Meta;
