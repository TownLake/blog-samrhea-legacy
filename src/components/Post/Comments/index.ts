export { default as Comments } from "./Comments";
