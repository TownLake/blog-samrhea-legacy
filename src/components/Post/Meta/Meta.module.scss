@import "../../../assets/scss/variables";
@import "../../../assets/scss/mixins";

.meta {
  .date {
    font-style: italic;
  }
}
