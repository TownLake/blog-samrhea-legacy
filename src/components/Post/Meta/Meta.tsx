import React from "react";

import * as styles from "./Meta.module.scss";

interface Props {
  date: string;
}

const Meta: React.FC<Props> = ({ date }: Props) => (
  <div className={styles.meta}>
    <p className={styles.date}>
      Published{" "}
      {new Date(date).toLocaleDateString("en-US", {
        year: "numeric",
        month: "short",
        day: "numeric",
      })}
    </p>
  </div>
);

export default Meta;
