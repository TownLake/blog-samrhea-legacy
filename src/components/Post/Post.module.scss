@import "src/assets/scss/variables";
@import "src/assets/scss/mixins";

.post {
  .content {
    @include margin-auto();
  }

  .footer {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .comments {
    max-width: $layout-post-width;
    @include padding(0, 0.5);
    @include margin-auto();
  }

  .buttons {
    align-items: center;
    display: flex;
    justify-content: center;
    text-align: center;
    @include margin(1, 0, 0, 0);

    .buttonArticles {
      @include margin(0, 0.5, 0, 0);
    }
  }
}

@include breakpoint-md {
  .post {
    .footer {
      @include padding-equal(0);
    }

    .comments {
      @include padding-equal(0);
    }

    .buttons {
      left: 30px;
      max-width: none;
      position: fixed;
      top: 30px;
      @include margin(0, 0.5, 0, 0);
    }
  }
}
