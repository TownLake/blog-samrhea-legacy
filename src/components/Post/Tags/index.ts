export { default as Tags } from "./Tags";
