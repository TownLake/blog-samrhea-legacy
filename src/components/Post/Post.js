// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import Author from './Author';
import Comments from './Comments';
import Content from './Content';
import Meta from './Meta';
import Tags from './Tags';
import styles from './Post.module.scss';
import type { Node } from '../../types';

type Props = {
  post: Node
};

const Post = ({ post }: Props) => {
  const { html } = post;
  const { tagSlugs, slug } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles['post']}>
      <Link className={styles['post__home-button']} to="/">Home</Link>

      <div className={styles['post__content']}>
        <Content body={html} title={title} />
      </div>

      <div className={styles['post__footer']}>
        <Meta date={date} />
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles['post__comments']}>
        <Comments postSlug={slug} postTitle={post.frontmatter.title} />
      </div>
    </div>
  );
};

export default Post;
