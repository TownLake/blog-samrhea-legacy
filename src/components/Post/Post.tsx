import React from "react";

import { Button } from "@/components/Button";
import { ThemeSwitcher } from "@/components/ThemeSwitcher";
import type { Node } from "@/types";

import { Author } from "./Author";
import { Comments } from "./Comments";
import { Content } from "./Content";
import { Meta } from "./Meta";
import { Tags } from "./Tags";

import * as styles from "./Post.module.scss";

interface Props {
  post: Node;
}

const Post: React.FC<Props> = ({ post }: Props) => {
  const { html } = post;
  const { tagSlugs, slug } = post.fields;
  const { tags, title, date } = post.frontmatter;

  return (
    <div className={styles.post}>
      <div className={styles.buttons}>
        <Button className={styles.buttonArticles} title="All Articles" to="/" />
        <ThemeSwitcher />
      </div>

      <div className={styles.content}>
        <Content body={html} title={title} />
      </div>

      <div className={styles.footer}>
        <Meta date={date} />
        {tags && tagSlugs && <Tags tags={tags} tagSlugs={tagSlugs} />}
        <Author />
      </div>

      <div className={styles.comments}>
        <Comments postSlug={slug} postTitle={post.frontmatter.title} />
      </div>
    </div>
  );
};

export default Post;
