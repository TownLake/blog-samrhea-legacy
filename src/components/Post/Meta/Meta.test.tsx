import React from "react";

import { Meta } from "@/components/Post/Meta";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

describe("Meta", () => {
  test("renders correctly", () => {
    const props = {
      date: mocks.markdownRemark.frontmatter.date,
    };

    const tree = testUtils
      .createSnapshotsRenderer(<Meta {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
