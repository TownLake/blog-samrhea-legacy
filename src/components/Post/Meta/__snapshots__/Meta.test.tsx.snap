// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta renders correctly 1`] = `
<div>
  <p>
    Published
     
    Sep 1, 2016
  </p>
</div>
`;
